controllers/updates/index.js

```diff
@@ -0,0 +1,125 @@
+'use strict';
+const path = require('path');
+const express = require('express');
+const { concat, isArray, pick, get } = require('lodash');
+
+const router = express.Router();
+
+const { injectBreadcrumbs } = require('../../middleware/inject-content');
+const { getUpdates } = require('../../services/content-api');
+const appData = require('../../modules/appData');
+const { buildPagination } = require('../../modules/pagination');
+
+function renderListing({ res, title, entries = [], entriesMeta = null, breadcrumbs = [], isNewsLandingPage = false }) {
+    const pagination = buildPagination(entriesMeta.pagination);
+    res.render(path.resolve(__dirname, './views/listing'), {
+        title,
+        entries,
+        entriesMeta,
+        pagination,
+        breadcrumbs,
+        isNewsLandingPage
+    });
+}
+
+if (appData.isNotProduction) {
+    router.get('/:type?/:date?/:slug?', injectBreadcrumbs, async (req, res, next) => {
+        let urlPath = '';
+        let query = {};
+
+        const isSinglePost = req.params.date && req.params.slug;
+
+        if (req.params.type) {
+            // Due to quirks of Craft's entry types,
+            // we have to convert snake_case into kebab-case
+            // both at the API end and then in reverse here.
+            urlPath += `/${req.params.type.replace('-', '_')}`;
+        }
+
+        if (isSinglePost) {
```

Comment:
Is there a way of rewriting this so the URL path code is one block of logic and the query is another? Would be nice to collect related logic together.

Reply:
Yeah, have simplified this now.